### DISCUSSION TITLE
What is the average orientation?

### DISCUSSION THREAD

**Discussion by Brian-Liu-rx** [mtex-toolbox/mtex#2017] [Ask Anything]
Hi, everyone,
    If I have some orientations, how can we calculate the average orientation? I know that in mtex we have "grain.meanOrientation" but what is the theory of average orientation?
    Thanks!
    Ruxue Liu

**Comment by zmichels** [comment]
You can use the function: mean()


See the details here for full syntax and options:

https://mtex-toolbox.github.io/orientation.mean.html

**Reply by Brian-Liu-rx** [reply]
Hi, zmichels,
    Thanks for your reply! Now I know how to do it.
    And I also want to understand the mathematical mechanism behind it, namely the equation to calculate the mean value of two or more rotation matrixes.
    Regards,
    Brian Liu

**Reply by Brian-Liu-rx** [reply]
@zmichels

**Reply by zmichels** [reply]
FWIW, I think the specific implementation might get better use quaternion representations as opposed to rotation matrices. I could be very wrong tho.

**Reply by Brian-Liu-rx** [reply]
quaternion is also ok, any mathematical  expression is ok for me :)

**Reply by zmichels** [reply]
Oh whew... as I find quaternion very nonhuman to work with personally.

Here's a reference I came across on the topic:

[1] Markley, F. Landis, Yang Chen, John Lucas Crassidis, and Yaakov Oshman. "Average Quaternions." Journal of Guidance, Control, and Dynamics. Vol. 30, Issue 4, 2007, pp. 1193-1197.

**Reply by Brian-Liu-rx** [reply]
Thank a lot! zmichels
